pkg/sql/zone_config.go

```diff
@@ -189,17 +194,30 @@ func zoneConfigHook(
 	cfg *config.SystemConfig, id config.SystemTenantObjectID,
 ) (*zonepb.ZoneConfig, *zonepb.ZoneConfig, bool, error) {
 	getKey := func(key roachpb.Key) (*roachpb.Value, error) {
+		// TODO(arul): clean this up
```

Comment:
Huh, why did we need this? 

I think generally keeping around this `SystemTenantObjectID` type, compared to other instances where we're passing in the codec + desc ID, and me not fully being able to appreciate the future of this `SystemConfig` type becoming makes some of these invariants non-obvious to me. 

This is pretty hairy code, and trying to untangle it to make the system tenant less distinguished is difficult, but I'd benefit from a high-level direction you see these type taking. Is that apparent to you? No worries if not, just wanted to share the confusion. A lot of the "host tenant only" stuff is in the kvserver/reports/reporter, which will also later end up generalizing to tenants.